### ISSUE TITLE
Support exact equality on some fields

### ISSUE THREAD

**Issue by Sufflope** [jonaspleyer/approx-derive#3]
Hello, just like you can currently `skip` fields, for my use case I would need some fields to be compared with `==`. Example case would be a struct like:

```rust
struct Prediction {
    category: String,
    confidence: f64
}
```

where I want approx equality for `confidence` but exact equality for `category`.

I'm taking the opportunity to say that unfortunately, between my first usage of `approx` 2 years ago when I didn't feel the need for a derive, and the last one this summer where the urge kicked in, I didn't notice that you started this project, so I started mine :sweat_smile:. Maybe you could be interested in having a look : https://github.com/brendanzab/approx/pull/89

I think you might be interested in `darling` (e.g. the problem solved by https://github.com/jonaspleyer/approx-derive/pull/1 would not happen with it instead of manual (and tedious IMO) parsing).

I took a few diverging design options based on my use-case which seems a bit different than yours, but maybe we can work something out?

**Comment by jonaspleyer** [comment]
Hi and thanks for taking the time to give this feedback. Comparing with the `==` operator makes sense as you point out. I will look into this and add it as soon as possible.

Concerning the integration into mainline: I think from a practical perspective it is much more favorable to have the derive macros under a feature flag in the main crate. I will have a look and see if we can reuse some of my code there.

I have come across `darling` in the past but was occupied with too much other work to truly give it a go. If I find the time and feel the need to to do, I will come back to this. Thanks for pointing this out to me though!

**Comment by jonaspleyer** [comment]
I have added this new feature and published a version `0.1.5` on crates.io. You should be able to use it from now on. Please check the documentation and thank you for this suggested addition. Your example would have to be implemented like so:
```rust
// Note that we need to specify the epsilon_type = ... 
// since the type gets inferred from the first field type otherwise.
#[derive(AbsDiffEq, PartialEq, Debug)]
#[approx(epsilon_type = f64)]
struct Prediction {
    #[approx(equal)]
    category: String,
    confidence: f64,
}

// Or if you can change the order of the fields:
#[derive(AbsDiffEq, PartialEq, Debug)]
struct Prediction {
    confidence: f64,
    #[approx(equal)]
    category: String,
}
```
These cases have been included in the test-suite. See https://github.com/jonaspleyer/approx-derive/blob/main/tests/derive_abs_diff.rs#L144

**Comment by jonaspleyer** [comment]
*Update*
I just had the idea to add a new feature which allows you to map a value to another before comparing them. Please have a look at the new documentation:
https://docs.rs/approx-derive/latest/approx_derive/#mapping-values
